ptstats/ptstats_test.go

```diff
@@ -71,337 +67,459 @@ func (da *TestCountySubdivisionDataAccess) GetStateFp(countyFp string) string {
-	assert.Equal(oldCousub.PopulationMale, s.SynthCousubStatsDA.GetMalePopulation(countyFp, cousubFp), "Subdivision male population should not change")
-	assert.Equal(oldCousub.PopulationFemale-1, s.SynthCousubStatsDA.GetFemalePopulation(countyFp, cousubFp), "Subdivision female population should decrement by 1")
-	newCousubPopPerSqMile = float64(oldCousub.Population-1) / sqMiles
-	assert.Equal(newCousubPopPerSqMile, s.SynthCousubStatsDA.GetPopulationPerSquareMile(countyFp, cousubFp), "Subdivision population density should change with an decrease in population")
-
-	// 5. Test that stats don't change for a body missing patient city
-	// ===============================================================
-	body = buildPatientRequestObject("female", "")
-	cousubFp = s.CousubDA.GetCountySubdivisionFp("")
-	countyFp = s.CousubDA.GetCountyFp(cousubFp)
-
-	req, _ = http.NewRequest("POST", "http://example.com/Patient", ioutil.NopCloser(bytes.NewReader(body)))
-	c.Request = req
-	testStatsDontChange(t, s, c, countyFp, cousubFp)
-
-	// 6. Test that stats don't change for a body missing patient gender
-	// =================================================================
-	body = buildPatientRequestObject("", "Boston")
-	cousubFp = s.CousubDA.GetCountySubdivisionFp("Boston")
-	countyFp = s.CousubDA.GetCountyFp(cousubFp)
-
-	req, _ = http.NewRequest("POST", "http://example.com/Patient", ioutil.NopCloser(bytes.NewReader(body)))
-	c.Request = req
-	testStatsDontChange(t, s, c, countyFp, cousubFp)
-
-	// 7. Test that stats don't change for HTTP methods that aren't POST or DELETE
-	// ===========================================================================
-	body = buildPatientRequestObject("male", "Boston")
-	methods := [6]string{"GET", "PUT", "HEAD", "CONNECT", "OPTIONS", "TRACE"}
-
-	for _, method := range methods {
-		req, _ = http.NewRequest(method, "http://example.com/Patient", ioutil.NopCloser(bytes.NewReader(body)))
-		c.Request = req
-		testStatsDontChange(t, s, c, countyFp, cousubFp)
-	}
+	assert.Equal(oldCousub.Population-1, newCousub.Population, "Subdivision population should decrement by 1")
+	assert.Equal(oldCousub.PopulationMale, newCousub.PopulationMale, "Subdivision male population should not change")
+	assert.Equal(oldCousub.PopulationFemale-1, newCousub.PopulationFemale, "Subdivision female population should decrement by 1")
+	newCousubPopPerSqMile = float64(oldCousub.Population-1) / 5.0
+	assert.Equal(newCousubPopPerSqMile, newCousub.PopulationPerSquareMile, "Subdivision population density should change with an decrease in population")
 }
 
-// Reusable testing submodule that verifies no patient statistics have changed
-func testStatsDontChange(t *testing.T, s *PtStatsInterceptor, c *gin.Context, countyFp, cousubFp string) {
+// TestPatientStatsInterceptorErrorHandling tests that the interceptors and underlying
+// data access layers handle errors as expected
+func TestPatientStatsInterceptorErrorHandling(t *testing.T) {
```

Comment:
While refactoring the ptstats module (working on the diabetes-stats branch) I realized I never implemented these tests to check the error handling of the interceptors. Since interceptors fail silently these tests seem less important than the others but I'll implement them when writing tests for the diabetes-stats interceptors.